### DISCUSSION TITLE
MedicalSkillGainSpeed and WeaponsSkillGainSpeed are mixed up

### DISCUSSION THREAD

**Discussion by bananaman6213** [FakeFishGames/Barotrauma#14444] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

MedicalSkillGainSpeed (used for Journeyman assistants apprenticed to medics) makes weapons level faster, while WeaponsSkillGainSpeed (used by captain's Deputy/Journeyman assistants apprenticed to security) makes medical level faster

bug seems to be caused by these lines in CharacterInfo.cs, at about line 1320
![image](https://github.com/user-attachments/assets/69b55851-4b08-4446-8769-995a37e0d95c)

https://github.com/user-attachments/assets/8d4dbff8-d983-4042-a1ef-293f431315f9

### Reproduction steps

1. Start a game and spawn an assistant
2. Either give yourself a medic/security apprenticeship and the Journeyman talent, or get a captain with Deputy
3. Observe how books give more experience to the wrong skill

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report! Fixed in https://github.com/FakeFishGames/Barotrauma-development/commit/c14eac64d13403e8942cdcc43c729aed15b16e6c (commit in our private repo). Fix coming up in the next release.
